common/janitor.c

```diff
@@ -572,15 +607,38 @@ handlesignal(int signal, siginfo_t* info, void* data)
     }
 }
 
+void
+janitor_backtrace(void)
+{
+#ifdef HAVE_BACKTRACE_FULL
+    if(frames == NULL) {
+        frames = backtrace_create_state(NULL, 0, (backtrace_error_callback) errorhandler, NULL);
+    }
+    pthread_mutex_lock(&frameslock);
+    outputbacktrace(1, frames);
+    pthread_mutex_unlock(&frameslock);
+#else
+    outputbacktrace(1, NULL);
+#endif
+}
+
+void
+janitor_backtrace_all(void)
+{
+    dumpthreads();
+}
+
 int
 janitor_trapsignals(char* argv0)
 {
     sigset_t mask;
     stack_t ss;
     struct sigaction newsigaction;
+    static struct backtrace_state *frames;
 
 #ifdef HAVE_BACKTRACE_FULL
-    CHECKFAIL((state = backtrace_create_state(argv0, 0, errorhandler, NULL)) == NULL);
+    CHECKFAIL((state  = backtrace_create_state(argv0, 0, (backtrace_error_callback)errorhandler, NULL)) == NULL);
+    CHECKFAIL((frames = backtrace_create_state(argv0, 0, (backtrace_error_callback)errorhandler, NULL)) == NULL);
```

Comment:
Not at the moment, overlooked remnant of other code that I could not get to work and was mainly useful for debugging purpose for now.  The janitor_backtrace_string() call that would dump your current stack trace into a string.  Handly when for instance a task is scheduled due to some procuedure (which you want to find out) and later on executed or removed, but at that time you want to know when and how it was created.  But that wasn't portable enough yet, so I ditched that code, but this line was left.  I'll remove it.